ci/test/substitutions

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -ex
+# shellcheck source=./ci/test/common
+source ./ci/test/common
+init_test_repo
+cd "$test_repo_path"
+
+cat <<EOF > main.workflow
+workflow "wf" {
+  resolves = "b"
+}
+
+action "b" {
+  uses = "_VAR1"
```

Comment:
Hi, @ivotron, Sorry for the delay, Actually there are two issues arising after using the dollar symbol as the starting symbol first the substitutions arguments need to be passed as a string second. Second when I used the $ symbol in sample workflows of integration tests then they are not being parsed in the workflow so I decided to remove $ symbol from the beginning of the variables.

Example to use the substitutions feature : 

```popper run --substitutions _variable=v1 --substitutions _variable2=v2```

Reply:
> after using the dollar symbol as the starting symbol first the substitutions arguments need to be passed as a string

can you please elaborate on the above? I am not sure I understand your question

Reply:
Actually, to show in a better way, I am attaching screenshots for the same home it helps out.
There are based on what I assumed from your previous comment.
If I use dollar in substitutions as well
![Screenshot from 2020-01-22 06-44-40](https://user-images.githubusercontent.com/18120004/72857667-4ff6b600-3ce4-11ea-93d8-3ddd05d28f11.jpg)

The substitutions list becomes blank

![Screenshot from 2020-01-22 06-44-25](https://user-images.githubusercontent.com/18120004/72857692-7157a200-3ce4-11ea-8d2b-d9a7bb5ee02e.jpg)

But when I passed it in the form of string
![Screenshot from 2020-01-22 06-47-37](https://user-images.githubusercontent.com/18120004/72857720-82a0ae80-3ce4-11ea-86ee-4519ea4b495e.jpg)

Then the substitutions list got passed but then workflow variables turned blank.
![Screenshot from 2020-01-22 06-45-55](https://user-images.githubusercontent.com/18120004/72857779-ad8b0280-3ce4-11ea-8962-a95c51ac4f31.jpg)

But from your last comment If I use it in only workflow file then I would still obtain them blank during parsing.

Reply:
@vipulchhabra99, I see where the confusion is coming from. The `$` is only used in the `.workflow` file; it is **NOT** used when passed in the `--substitutions` flag. So this:

```hcl
workflow "foo" {
  resolves = "$_VAR"
}
```

is defined as:

```bash
popper run --substitution VAR=foo
```

btw, could we do `--substitution` (singular) instead of `--substitutions`? This is more appropriate since each flag defines a single substitution.

Reply:
Yeah actually substitution seems a better one. If i use $ in workflow the variable disappears after parsing. In the last screen shot I am printing parsed workflow. Please check that once.

Reply:
we need to escape the dollar sign if we're using it in a script (`\$`)

Reply:
Yeah, thanks it worked.